### PR TITLE
feat: introducing a new smart input hoppscotch ui component

### PR DESCRIPTION
**Pull request by JoelJacobStephen**
### Issue Number
Closes HFE-61

### Description
This PR introduces a new Smart Input component in `hoppscotch-ui` where most of the input implementation in the different modules is abstracted into the new UI component. This includes input boxes with floating labels, input with buttons, single input boxes etc..

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: packages/hoppscotch-ui/src/components/smart/Input.vue [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  return `input-${s4()}${s4()}`
+}
+
+const props = withDefaults(
+  defineProps<{
+    id: string
+    styles: string
+    modelValue: string | null
+    placeholder: string
+    inputStyles: any
```
Comment (anwarulislam) [inline packages/hoppscotch-ui/src/components/smart/Input.vue:41]: The type shouldn't be `any`

File: packages/hoppscotch-ui/src/components/smart/Input.vue [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    inputType: string | (string | false)[]
```
Comment (anwarulislam) [inline packages/hoppscotch-ui/src/components/smart/Input.vue:42]: Why is this `inputType` prop for? I can't see any usage in the component.

**Review by anwarulislam — CHANGES_REQUESTED** [review]
@JoelJacobStephen please look at these two comments.

**Review by anwarulislam — APPROVED** [review]
LGTM 🚀

File: packages/hoppscotch-sh-admin/src/components/users/InviteModal.vue [hunk 3]
```diff
@@ -6,19 +6,11 @@
...
-          placeholder=" "
-          type="text"
-          autocomplete="off"
-          @keyup.enter="sendInvite"
-        />
-        <label for="inviteUserEmail">{{ t('users.email_address') }}</label>
-      </div>
+      <HoppSmartInput
+        v-model="email"
+        label="Email Address"
```
Comment (nivedin) [inline packages/hoppscotch-sh-admin/src/components/users/InviteModal.vue:11]:
```suggestion
        :label="t('users.email_address')"
```

File: packages/hoppscotch-common/src/pages/profile.vue [hunk 4]
```diff
@@ -251,11 +240,13 @@ const loadingCurrentUser = computed(() => {
...
 })
 
-const displayName = ref(currentUser.value?.displayName)
+const displayName = ref(currentUser.value?.displayName || "")
 const updatingDisplayName = ref(false)
-watchEffect(() => (displayName.value = currentUser.value?.displayName))
+watchEffect(() => (displayName.value = currentUser.value?.displayName || ""))
 
 const updateDisplayName = () => {
+  console.log(displayName.value)
```
Comment (nivedin) [inline packages/hoppscotch-common/src/pages/profile.vue:248]: remove console statement

**Review by nivedin — CHANGES_REQUESTED** [review]
please look into these comments

**Pull request merged**